tests/test_simulated_auxiliary.py

```diff
@@ -95,6 +105,127 @@ def prepare_config():
     return config_file_path
 
 
+@pytest.fixture
+def mock_channel(mocker):
+    class MockProxyCChannel(CChannel):
+        def __init__(self, name=None, *args, **kwargs):
+            self.name = name
+            self.queue_in = queue.Queue()
+            self.queue_out = queue.Queue()
+            super(MockProxyCChannel, self).__init__(*args, **kwargs)
+
+        _cc_open = mocker.stub(name="_cc_open")
+        _cc_close = mocker.stub(name="_cc_close")
+        _cc_send = mocker.stub(name="_cc_send")
+        _cc_receive = mocker.stub(name="_cc_receive")
+
+    return MockProxyCChannel(name="test_aux")
+
+
+@pytest.fixture
+def simulated_constructor_init(mocker, mock_channel, mock_msg):
+    class Dict_modified(dict):
```

Comment:
class not necessary ?
Pytest gives you the tools.
mocker.patch.object(simulation.Simulation, "get_scenario", return_value=scenario.Scenario([None]))